# Statistical Analysis of Customer Shopping Behavior

Import necessary python libraries and data 

In [1]:
import pandas as pd
import numpy as np
import statsmodels.api as sm
from statsmodels.formula.api import ols
import scipy as scipy
import itertools
campaign_data = pd.read_csv('customer_shopping_behavior.csv')

In [2]:
campaign_data.head(10)

,Customer ID,Age,Gender,Item Purchased,Category,Purchase Amount (USD),Location,Size,Color,Season,Review Rating,Subscription Status,Shipping Type,Discount Applied,Promo Code Used,Previous Purchases,Payment Method,Frequency of Purchases
0,1,55,Male,Blouse,Clothing,53,Kentucky,L,Gray,Winter,3.1,Yes,Express,Yes,Yes,14,Venmo,Fortnightly
1,2,19,Male,Sweater,Clothing,64,Maine,L,Maroon,Winter,3.1,Yes,Express,Yes,Yes,2,Cash,Fortnightly
2,3,50,Male,Jeans,Clothing,73,Massachusetts,S,Maroon,Spring,3.1,Yes,Free Shipping,Yes,Yes,23,Credit Card,Weekly
3,4,21,Male,Sandals,Footwear,90,Rhode Island,M,Maroon,Spring,3.5,Yes,Next Day Air,Yes,Yes,49,PayPal,Weekly
4,5,45,Male,Blouse,Clothing,49,Oregon,M,Turquoise,Spring,2.7,Yes,Free Shipping,Yes,Yes,31,PayPal,Annually
5,6,46,Male,Sneakers,Footwear,20,Wyoming,M,White,Summer,2.9,Yes,Standard,Yes,Yes,14,Venmo,Weekly
6,7,63,Male,Shirt,Clothing,85,Montana,M,Gray,Fall,3.2,Yes,Free Shipping,Yes,Yes,49,Cash,Quarterly
7,8,27,Male,Shorts,Clothing,34,Louisiana,L,Charcoal,Winter,3.2,Yes,Free Shipping,Yes,Yes,19,Credit Card,Weekly
8,9,26,Male,Coat,Outerwear,97,West Virginia,L,Silver,Summer,2.6,Yes,Express,Yes,Yes,8,Venmo,Annually
9,10,57,Male,Handbag,Accessories,31,Missouri,M,Pink,Spring,4.8,Yes,2-Day Shipping,Yes,Yes,4,Cash,Quarterly


In [3]:
campaign_data.duplicated().sum()

np.int64(0)

In [4]:
campaign_data.columns = campaign_data.columns.str.lower().str.replace(' ', '_')

In [5]:
campaign_data.describe(include='all')

,customer_id,age,gender,item_purchased,category,purchase_amount_(usd),location,size,color,season,review_rating,subscription_status,shipping_type,discount_applied,promo_code_used,previous_purchases,payment_method,frequency_of_purchases
count,3900.000000,3900.000000,3900,3900,3900,3900.000000,3900,3900,3900,3900,3863.000000,3900,3900,3900,3900,3900.000000,3900,3900
unique,NaN,NaN,2,25,4,NaN,50,4,25,4,NaN,2,6,2,2,NaN,6,7
top,NaN,NaN,Male,Blouse,Clothing,NaN,Montana,M,Olive,Spring,NaN,No,Free Shipping,No,No,NaN,PayPal,Every 3 Months
freq,NaN,NaN,2652,171,1737,NaN,96,1755,177,999,NaN,2847,675,2223,2223,NaN,677,584
mean,1950.500000,44.068462,NaN,NaN,NaN,59.764359,NaN,NaN,NaN,NaN,3.750065,NaN,NaN,NaN,NaN,25.351538,NaN,NaN
std,1125.977353,15.207589,NaN,NaN,NaN,23.685392,NaN,NaN,NaN,NaN,0.716983,NaN,NaN,NaN,NaN,14.447125,NaN,NaN
min,1.000000,18.000000,NaN,NaN,NaN,20.000000,NaN,NaN,NaN,NaN,2.500000,NaN,NaN,NaN,NaN,1.000000,NaN,NaN
25%,975.750000,31.000000,NaN,NaN,NaN,39.000000,NaN,NaN,NaN,NaN,3.100000,NaN,NaN,NaN,NaN,13.000000,NaN,NaN
50%,1950.500000,44.000000,NaN,NaN,NaN,60.000000,NaN,NaN,NaN,NaN,3.800000,NaN,NaN,NaN,NaN,25.000000,NaN,NaN
75%,2925.250000,57.000000,NaN,NaN,NaN,81.000000,NaN,NaN,NaN,NaN,4.400000,NaN,NaN,NaN,NaN,38.000000,NaN,NaN


In [6]:
campaign_data.isnull().sum()

customer_id                0
age                        0
gender                     0
item_purchased             0
category                   0
purchase_amount_(usd)      0
location                   0
size                       0
color                      0
season                     0
review_rating             37
subscription_status        0
shipping_type              0
discount_applied           0
promo_code_used            0
previous_purchases         0
payment_method             0
frequency_of_purchases     0
dtype: int64

In [7]:
campaign_data.head()

,customer_id,age,gender,item_purchased,category,purchase_amount_(usd),location,size,color,season,review_rating,subscription_status,shipping_type,discount_applied,promo_code_used,previous_purchases,payment_method,frequency_of_purchases
0,1,55,Male,Blouse,Clothing,53,Kentucky,L,Gray,Winter,3.1,Yes,Express,Yes,Yes,14,Venmo,Fortnightly
1,2,19,Male,Sweater,Clothing,64,Maine,L,Maroon,Winter,3.1,Yes,Express,Yes,Yes,2,Cash,Fortnightly
2,3,50,Male,Jeans,Clothing,73,Massachusetts,S,Maroon,Spring,3.1,Yes,Free Shipping,Yes,Yes,23,Credit Card,Weekly
3,4,21,Male,Sandals,Footwear,90,Rhode Island,M,Maroon,Spring,3.5,Yes,Next Day Air,Yes,Yes,49,PayPal,Weekly
4,5,45,Male,Blouse,Clothing,49,Oregon,M,Turquoise,Spring,2.7,Yes,Free Shipping,Yes,Yes,31,PayPal,Annually


In [8]:
campaign_data["review_rating"]=campaign_data.groupby("category")["review_rating"].transform(lambda x: x.fillna(x.mean()))


In [9]:
label=['Young Adults', 'Adults', 'Middle Aged', 'Seniors']
campaign_data['age_group'] = pd.cut(campaign_data['age'], bins=[18, 25, 40, 60, 100], labels=label)

In [10]:
campaign_data[["age_group", "age"]].head(10)

,age_group,age
0,Middle Aged,55
1,Young Adults,19
2,Middle Aged,50
3,Young Adults,21
4,Middle Aged,45
5,Middle Aged,46
6,Seniors,63
7,Adults,27
8,Adults,26
9,Middle Aged,57


In [11]:
frequency_mapping = {"Fortnightly": 14, "Monthly": 30, "Weekly": 7, "Daily": 7,"Quarterly": 90, "Yearly": 365,"Bi-Weekly": 14,"Every 3 Months": 90,"Every 6 Months": 180,"Every 2 Weeks": 14,"Every 4 Weeks": 28,"Every 8 Weeks": 56,"Every 12 Weeks": 84}
campaign_data["purchase_frequency"] = campaign_data["frequency_of_purchases"].map(frequency_mapping)

In [12]:
campaign_data[["purchase_frequency", "frequency_of_purchases"]].head(10)

,purchase_frequency,frequency_of_purchases
0,14.0,Fortnightly
1,14.0,Fortnightly
2,7.0,Weekly
3,7.0,Weekly
4,NaN,Annually
5,7.0,Weekly
6,90.0,Quarterly
7,7.0,Weekly
8,NaN,Annually
9,90.0,Quarterly


In [13]:
campaign_data[["discount_applied", "promo_code_used"]].head(10)

,discount_applied,promo_code_used
0,Yes,Yes
1,Yes,Yes
2,Yes,Yes
3,Yes,Yes
4,Yes,Yes
5,Yes,Yes
6,Yes,Yes
7,Yes,Yes
8,Yes,Yes
9,Yes,Yes


In [14]:
(campaign_data["discount_applied"]==campaign_data["promo_code_used"]).all()

np.True_

In [15]:
campaign_data.drop(columns=["promo_code_used"], inplace=True)

In [16]:
campaign_data["previous_purchases_category"]=campaign_data["previous_purchases"].transform( lambda x: "New" if x<10 else ("Returning" if x<50 else "Loyal")) 

In [17]:
campaign_data.columns

Index(['customer_id', 'age', 'gender', 'item_purchased', 'category',
       'purchase_amount_(usd)', 'location', 'size', 'color', 'season',
       'review_rating', 'subscription_status', 'shipping_type',
       'discount_applied', 'previous_purchases', 'payment_method',
       'frequency_of_purchases', 'age_group', 'purchase_frequency',
       'previous_purchases_category'],
      dtype='object')

In [18]:
campaign_data.rename(columns={"purchase_amount_(usd)": "purchase_amount"}, inplace=True)

In [19]:
campaign_data.dtypes

customer_id                       int64
age                               int64
gender                           object
item_purchased                   object
category                         object
purchase_amount                   int64
location                         object
size                             object
color                            object
season                           object
review_rating                   float64
subscription_status              object
shipping_type                    object
discount_applied                 object
previous_purchases                int64
payment_method                   object
frequency_of_purchases           object
age_group                      category
purchase_frequency              float64
previous_purchases_category      object
dtype: object

In [20]:
campaign_data['age_group'] =campaign_data['age_group'].astype('object')
campaign_data.columns

Index(['customer_id', 'age', 'gender', 'item_purchased', 'category',
       'purchase_amount', 'location', 'size', 'color', 'season',
       'review_rating', 'subscription_status', 'shipping_type',
       'discount_applied', 'previous_purchases', 'payment_method',
       'frequency_of_purchases', 'age_group', 'purchase_frequency',
       'previous_purchases_category'],
      dtype='object')

ANOVA Analysis

In [21]:
independent_var=['gender','item_purchased','category','location','size','color','season','subscription_status','shipping_type','discount_applied','payment_method','previous_purchases_category']
dependent_var=['purchase_amount','review_rating','purchase_frequency']

In [22]:
result_table=[]
factors= itertools.combinations(independent_var,2)
i='purchase_amount'
for j in factors:
        model=ols(f'{i}~C({j[0]})+C({j[1]})+C({j[0]}):C({j[1]})', data=campaign_data).fit()
        result=sm.stats.anova_lm(model, typ=2)
        for k in result.index:
            if k != "Residual":
                result_table.append(pd.DataFrame([{"Dependent_variable":i,"Factor_1":k,
                                    "DF":result.loc[k,"df"],
                                    "F":result.loc[k,"F"],
                                    "P_Value":result.loc[k,"PR(>F)"]}]))
Final_Aova1=pd.concat(result_table, ignore_index=True)



C:\Users\Mr.Kaleeswaran\AppData\Local\Temp\ipykernel_18688\3170308748.py:5: SingularMatrixWarning: The design matrix is rank-deficient. The model parameters are not uniquely determined.
  model=ols(f'{i}~C({j[0]})+C({j[1]})+C({j[0]}):C({j[1]})', data=campaign_data).fit()
c:\Users\Mr.Kaleeswaran\AppData\Local\Programs\Python\Python312\Lib\site-packages\statsmodels\base\model.py:1988: ValueWarning: covariance of constraints does not have full rank. The number of constraints is 24, but rank is 2
  res = self.wald_test(
c:\Users\Mr.Kaleeswaran\AppData\Local\Programs\Python\Python312\Lib\site-packages\statsmodels\base\model.py:1988: ValueWarning: covariance of constraints does not have full rank. The number of constraints is 3, but rank is 0
  res = self.wald_test(
c:\Users\Mr.Kaleeswaran\AppData\Local\Programs\Python\Python312\Lib\site-packages\statsmodels\base\model.py:2140: RuntimeWarning: invalid value encountered in divide
  F /= J
c:\Users\Mr.Kaleeswaran\AppData\Local\Programs\Python\

In [23]:
result_table=[]
factors= itertools.combinations(independent_var,2)
i='review_rating'
campaign_data1=campaign_data.dropna()
for j in factors:
        model=ols(f'{i}~C({j[0]})+C({j[1]})+C({j[0]}):C({j[1]})', data=campaign_data1).fit()
        result=sm.stats.anova_lm(model, typ=2)
        for k in result.index:
            if k != "Residual":
                result_table.append(pd.DataFrame([{"Dependent_variable":i,"Factor_1":k,
                                    "DF":result.loc[k,"df"],
                                    "F":result.loc[k,"F"],
                                    "P_Value":result.loc[k,"PR(>F)"]}]))
Final_Aova2=pd.concat(result_table, ignore_index=True)

C:\Users\Mr.Kaleeswaran\AppData\Local\Temp\ipykernel_18688\1191379326.py:6: SingularMatrixWarning: The design matrix is rank-deficient. The model parameters are not uniquely determined.
  model=ols(f'{i}~C({j[0]})+C({j[1]})+C({j[0]}):C({j[1]})', data=campaign_data1).fit()
c:\Users\Mr.Kaleeswaran\AppData\Local\Programs\Python\Python312\Lib\site-packages\statsmodels\base\model.py:1988: ValueWarning: covariance of constraints does not have full rank. The number of constraints is 24, but rank is 2
  res = self.wald_test(
c:\Users\Mr.Kaleeswaran\AppData\Local\Programs\Python\Python312\Lib\site-packages\statsmodels\base\model.py:1988: ValueWarning: covariance of constraints does not have full rank. The number of constraints is 3, but rank is 0
  res = self.wald_test(
c:\Users\Mr.Kaleeswaran\AppData\Local\Programs\Python\Python312\Lib\site-packages\statsmodels\base\model.py:2140: RuntimeWarning: invalid value encountered in divide
  F /= J
c:\Users\Mr.Kaleeswaran\AppData\Local\Programs\Python

In [24]:
result_table=[]
factors= itertools.combinations(independent_var,2)
i='purchase_frequency'
for j in factors:
        model=ols(f'{i}~C({j[0]})+C({j[1]})+C({j[0]}):C({j[1]})', data=campaign_data).fit()
        result=sm.stats.anova_lm(model, typ=2)
        for k in result.index:
            if k != "Residual":
                result_table.append(pd.DataFrame([{"Dependent_variable":i,"Factor_1":k,
                                    "DF":result.loc[k,"df"],
                                    "F":result.loc[k,"F"],
                                    "P_Value":result.loc[k,"PR(>F)"]}]))
Final_Aova3=pd.concat(result_table, ignore_index=True)

C:\Users\Mr.Kaleeswaran\AppData\Local\Temp\ipykernel_18688\3049973143.py:5: SingularMatrixWarning: The design matrix is rank-deficient. The model parameters are not uniquely determined.
  model=ols(f'{i}~C({j[0]})+C({j[1]})+C({j[0]}):C({j[1]})', data=campaign_data).fit()
c:\Users\Mr.Kaleeswaran\AppData\Local\Programs\Python\Python312\Lib\site-packages\statsmodels\base\model.py:1988: ValueWarning: covariance of constraints does not have full rank. The number of constraints is 24, but rank is 2
  res = self.wald_test(
c:\Users\Mr.Kaleeswaran\AppData\Local\Programs\Python\Python312\Lib\site-packages\statsmodels\base\model.py:1988: ValueWarning: covariance of constraints does not have full rank. The number of constraints is 3, but rank is 0
  res = self.wald_test(
c:\Users\Mr.Kaleeswaran\AppData\Local\Programs\Python\Python312\Lib\site-packages\statsmodels\base\model.py:2140: RuntimeWarning: invalid value encountered in divide
  F /= J
c:\Users\Mr.Kaleeswaran\AppData\Local\Programs\Python\

In [25]:
Final_Aova1["Significance"]=Final_Aova1["P_Value"].apply(lambda x : "True" if x<0.05 else "False")
print(Final_Aova1.shape)

(198, 6)


In [26]:
Final_Aova2["Significance"]=Final_Aova2["P_Value"].apply(lambda x : "True" if x<0.05 else "False")
print(Final_Aova2.shape)

(198, 6)


In [27]:
Final_Aova3["Significance"]=Final_Aova3["P_Value"].apply(lambda x : "True" if x<0.05 else "False")
print(Final_Aova3.shape)

(198, 6)


In [28]:
for table_name, table in [('Purchase amount', Final_Aova1), ('Review rating', Final_Aova2), ('Purchase frequency', Final_Aova3)]:
    significant = table[table['P_Value'] < 0.05].sort_values('P_Value')
    print(f'\n{table_name}: {len(significant)} significant effects out of {len(table)} tests')
    print(significant[['Factor_1', 'F', 'P_Value']].to_string(index=False))


Purchase amount: 21 significant effects out of 198 tests
                      Factor_1        F      P_Value
                   C(location) 3.294334 3.047736e-11
                     C(season) 4.315824 4.798018e-03
                     C(season) 4.036180 7.078241e-03
                     C(season) 3.856444 9.079848e-03
                     C(season) 3.848720 9.177309e-03
                     C(season) 3.836065 9.340951e-03
                     C(season) 3.783234 1.004751e-02
                     C(season) 3.778629 1.011184e-02
                     C(season) 3.738177 1.069291e-02
                     C(season) 3.737941 1.069647e-02
                     C(season) 3.713233 1.106770e-02
                     C(season) 3.680459 1.157991e-02
C(previous_purchases_category) 3.452541 3.176217e-02
C(previous_purchases_category) 3.418688 3.285405e-02
C(previous_purchases_category) 3.405404 3.329228e-02
C(previous_purchases_category) 3.400665 3.345047e-02
C(previous_purchases_category) 3.393504 3

# Statistical Test Conclusion

## Test design

Three two-factor ANOVA result tables were produced for the customer-shopping data:

- `Final_Aova1`: purchase amount
- `Final_Aova2`: review rating
- `Final_Aova3`: purchase frequency, represented numerically as approximate days between purchases

For every pair of the 12 independent variables, the model tested both main effects and their interaction. Each table contains 198 tested terms. At the 5% significance level, an effect was treated as statistically significant when `P_Value < 0.05`; otherwise, there was insufficient evidence to reject the null hypothesis. The significance columns now correctly use the p-values from their own ANOVA table.

## ANOVA results

- **Purchase amount:** 21 of 198 terms were significant. The strongest recurring evidence involved `location` (`F = 3.2943`, `p < 0.001`), followed by effects involving `season` and `previous_purchases_category` (`p` values approximately 0.0048 to 0.0376). This indicates that spending varies across some locations, seasons, and loyalty groups, although the ANOVA table alone does not identify the specific levels that differ.
- **Review rating:** 24 of 198 terms were significant. `location` was the strongest result (`F = 6.5463`, `p < 0.001`). There was also evidence involving `size` and `color`, plus interactions between `category` and `payment_method`, and between `discount_applied` and `previous_purchases_category`. This suggests that ratings are associated with both product/customer grouping and some combined conditions, rather than one universal rating pattern.
- **Purchase frequency:** 4 of 198 terms were significant. The significant results were interactions between `shipping_type` and `previous_purchases_category` (`p = 0.0214`), `season` and `shipping_type` (`p = 0.0290`), `item_purchased` and `color` (`p = 0.0334`), and `size` and `discount_applied` (`p = 0.0490`). The interaction terms mean that the relationship with purchase frequency depends on the other factor; they should not be interpreted as standalone effects.

The ANOVA results show evidence of differences, but they do not specify which category levels differ. Because many terms were tested, individual p-values near 0.05 should be treated cautiously and supported with effect sizes, group means, and sample sizes.

## Tukey HSD conclusion

Tukey HSD was used as the post-hoc test at alpha = 0.05. The displayed filtered result contains three statistically significant pairwise differences:

- Purchase frequency differs between the `Green` and `Turquoise` colour groups (`p-adj = 0.0345`). The negative mean difference indicates that Green has a lower encoded frequency value than Turquoise. Since frequency labels were mapped to days, this is a difference in the encoded purchase interval, not automatically a difference in purchase count.
- Purchase frequency differs between `Fall` and `Spring` (`p-adj = 0.0407`), with Fall lower by approximately 2.82 encoded days.
- Purchase frequency differs between `Fall` and `Summer` (`p-adj = 0.0182`), with Fall lower by approximately 3.15 encoded days.

These Tukey results identify specific group pairs behind detectable frequency differences. No additional pairwise differences should be claimed from the filtered output. A non-significant comparison means that the data do not provide sufficient evidence of a difference at the 5% level; it does not prove that the group means are identical.

## Overall conclusion

Taken together, the tests indicate that customer spending, review ratings, and purchase frequency are not completely uniform across the observed customer, product, and transaction groups. Spending shows repeated differences associated with location, season, and previous-purchase segment. Ratings show strong location differences and additional size, colour, and interaction effects. Frequency has fewer omnibus findings, but its significant interactions show that purchasing intervals depend on combinations such as shipping type with loyalty segment or season.

The post-hoc evidence provides the clearest level-specific frequency findings: Fall has a shorter encoded purchase interval than Spring and Summer, and Green has a shorter encoded interval than Turquoise. These findings can guide further investigation of seasonal planning, colour-level merchandising, shipping experience, and loyalty-focused campaigns.

The results are associative, not causal. The data are observational, the frequency labels were converted into approximate day values, and unmeasured customer or product differences may explain part of the observed variation. Before using the results for production decisions, remove the five trailing artifact rows identified in the project report, use `.dropna()` consistently for both variables in every Tukey comparison, review group sizes and effect sizes, and validate promising interventions with dated transaction data or a controlled experiment.

In [29]:
turkey=[]
from statsmodels.stats.multicomp import pairwise_tukeyhsd

for i in dependent_var:
    for j in independent_var:
        print(campaign_data[[i,j]].dropna().dtypes)
        result=pairwise_tukeyhsd(endog=campaign_data[i], groups=campaign_data[j], alpha=0.05)
        result1=pd.DataFrame(result.summary().data[1:], columns=result.summary().data[0])
        result1.insert(0,"dependent_var",i)
        result1.insert(1,"Independent_var" ,j)
        turkey.append(result1)

final_tukey=pd.concat(turkey, ignore_index=True)

final_tukey["Significant"] = final_tukey["p-adj"].apply(lambda x: "Yes" if x < 0.05 else "No")  


purchase_amount     int64
gender             object
dtype: object
purchase_amount     int64
item_purchased     object
dtype: object
purchase_amount     int64
category           object
dtype: object
purchase_amount     int64
location           object
dtype: object


c:\Users\Mr.Kaleeswaran\AppData\Local\Programs\Python\Python312\Lib\site-packages\scipy\integrate\_quadpack_py.py:1286: IntegrationWarning: The integral is probably divergent, or slowly convergent.
  quad_r = quad(f, low, high, args=args, full_output=self.full_output,


purchase_amount     int64
size               object
dtype: object
purchase_amount     int64
color              object
dtype: object
purchase_amount     int64
season             object
dtype: object
purchase_amount         int64
subscription_status    object
dtype: object
purchase_amount     int64
shipping_type      object
dtype: object
purchase_amount      int64
discount_applied    object
dtype: object
purchase_amount     int64
payment_method     object
dtype: object
purchase_amount                 int64
previous_purchases_category    object
dtype: object
review_rating    float64
gender            object
dtype: object
review_rating     float64
item_purchased     object
dtype: object
review_rating    float64
category          object
dtype: object
review_rating    float64
location          object
dtype: object


c:\Users\Mr.Kaleeswaran\AppData\Local\Programs\Python\Python312\Lib\site-packages\scipy\integrate\_quadpack_py.py:1286: IntegrationWarning: The integral is probably divergent, or slowly convergent.
  quad_r = quad(f, low, high, args=args, full_output=self.full_output,


review_rating    float64
size              object
dtype: object
review_rating    float64
color             object
dtype: object
review_rating    float64
season            object
dtype: object
review_rating          float64
subscription_status     object
dtype: object
review_rating    float64
shipping_type     object
dtype: object
review_rating       float64
discount_applied     object
dtype: object
review_rating     float64
payment_method     object
dtype: object
review_rating                  float64
previous_purchases_category     object
dtype: object
purchase_frequency    float64
gender                 object
dtype: object
purchase_frequency    float64
item_purchased         object
dtype: object
purchase_frequency    float64
category               object
dtype: object
purchase_frequency    float64
location               object
dtype: object
purchase_frequency    float64
size                   object
dtype: object
purchase_frequency    float64
color                  object
dtype: obj

In [30]:
final_tukey[final_tukey["reject"]==True].reset_index()

,index,dependent_var,Independent_var,group1,group2,meandiff,p-adj,lower,upper,reject,Significant
0,1714,purchase_amount,color,Green,Turquoise,-10.0903,0.0345,-19.8908,-0.2899,True,Yes
1,1838,purchase_amount,season,Fall,Spring,-2.8192,0.0407,-5.5568,-0.0816,True,Yes
2,1839,purchase_amount,season,Fall,Summer,-3.1517,0.0182,-5.9202,-0.3831,True,Yes
